{"schemaVersion":{"minor":3,"patch":0,"major":0},"metadata":{"title":"AuthError","symbolKind":"enum","modules":[{"name":"Amplify"}],"externalID":"s:7Amplify9AuthErrorO","role":"symbol","fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AuthError"}],"navigatorTitle":[{"kind":"identifier","text":"AuthError"}],"roleHeading":"Enumeration"},"sections":[],"relationshipsSections":[{"identifiers":["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyError","doc:\/\/amplify-swift.Amplify\/s8CopyableP","doc:\/\/amplify-swift.Amplify\/s28CustomDebugStringConvertibleP","doc:\/\/amplify-swift.Amplify\/SQ","doc:\/\/amplify-swift.Amplify\/s5ErrorP","doc:\/\/amplify-swift.Amplify\/s8SendableP"],"kind":"relationships","title":"Conforms To","type":"conformsTo"}],"identifier":{"url":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError","interfaceLanguage":"swift"},"hierarchy":{"paths":[["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify"]]},"kind":"symbol","abstract":[{"type":"text","text":"Amplify error raised in the Auth category."}],"topicSections":[{"identifiers":["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/configuration(_:_:_:)","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/invalidState(_:_:_:)","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/notAuthorized(_:_:_:)","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/service(_:_:_:)","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/sessionExpired(_:_:_:)","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/signedOut(_:_:_:)","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/unknown(_:_:)","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/validation(_:_:_:_:)"],"generated":true,"title":"Enumeration Cases","anchor":"Enumeration-Cases"},{"title":"Default Implementations","generated":true,"identifiers":["doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/AmplifyError-Implementations","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/CustomDebugStringConvertible-Implementations","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/Equatable-Implementations","doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/Error-Implementations"],"anchor":"Default-Implementations"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/amplify\/autherror"]}],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["iOS"],"languages":["swift"],"tokens":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"text":"AuthError","kind":"identifier"}]}]}],"references":{"doc://amplify-swift.Amplify/s5ErrorP":{"type":"unresolvable","identifier":"doc:\/\/amplify-swift.Amplify\/s5ErrorP","title":"Swift.Error"},"doc://amplify-swift.Amplify/s8SendableP":{"type":"unresolvable","identifier":"doc:\/\/amplify-swift.Amplify\/s8SendableP","title":"Swift.Sendable"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/validation(_:_:_:_:)":{"abstract":[{"type":"text","text":"Caused when one of the input field is invalid"}],"title":"AuthError.validation(_:_:_:_:)","type":"topic","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/validation(_:_:_:_:)","url":"\/documentation\/amplify\/autherror\/validation(_:_:_:_:)","role":"symbol","fragments":[{"kind":"keyword","text":"case"},{"text":" ","kind":"text"},{"kind":"identifier","text":"validation"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:7Amplify5Fielda","text":"Field","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"ErrorDescription","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify16ErrorDescriptiona"},{"kind":"text","text":", "},{"kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona","text":"RecoverySuggestion"},{"kind":"text","text":", "},{"text":"Error","kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP"},{"text":"?)","kind":"text"}],"kind":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/service(_:_:_:)":{"role":"symbol","title":"AuthError.service(_:_:_:)","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/service(_:_:_:)","abstract":[{"text":"Caused by some error in the underlying service. Check the associated error for more details.","type":"text"}],"kind":"symbol","type":"topic","fragments":[{"text":"case","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"service"},{"kind":"text","text":"("},{"text":"ErrorDescription","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify16ErrorDescriptiona"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:7Amplify18RecoverySuggestiona","text":"RecoverySuggestion","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP","text":"Error"},{"text":"?)","kind":"text"}],"url":"\/documentation\/amplify\/autherror\/service(_:_:_:)"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/CustomDebugStringConvertible-Implementations":{"title":"CustomDebugStringConvertible Implementations","type":"topic","role":"collectionGroup","kind":"article","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/CustomDebugStringConvertible-Implementations","abstract":[],"url":"\/documentation\/amplify\/autherror\/customdebugstringconvertible-implementations"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/configuration(_:_:_:)":{"url":"\/documentation\/amplify\/autherror\/configuration(_:_:_:)","role":"symbol","abstract":[{"text":"Caused by issue in the way auth category is configured","type":"text"}],"fragments":[{"text":"case","kind":"keyword"},{"kind":"text","text":" "},{"text":"configuration","kind":"identifier"},{"text":"(","kind":"text"},{"text":"ErrorDescription","preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier"},{"kind":"text","text":", "},{"text":"RecoverySuggestion","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona"},{"kind":"text","text":", "},{"kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP","text":"Error"},{"kind":"text","text":"?)"}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/configuration(_:_:_:)","kind":"symbol","type":"topic","title":"AuthError.configuration(_:_:_:)"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/Equatable-Implementations":{"title":"Equatable Implementations","kind":"article","type":"topic","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/Equatable-Implementations","url":"\/documentation\/amplify\/autherror\/equatable-implementations","abstract":[],"role":"collectionGroup"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/sessionExpired(_:_:_:)":{"abstract":[{"type":"text","text":"Caused when a session is expired and needs the user to be re-authenticated"}],"type":"topic","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"sessionExpired"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier","text":"ErrorDescription"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona","text":"RecoverySuggestion"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"},{"text":"?)","kind":"text"}],"title":"AuthError.sessionExpired(_:_:_:)","role":"symbol","url":"\/documentation\/amplify\/autherror\/sessionexpired(_:_:_:)","kind":"symbol","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/sessionExpired(_:_:_:)"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/Error-Implementations":{"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/Error-Implementations","url":"\/documentation\/amplify\/autherror\/error-implementations","title":"Error Implementations","type":"topic","abstract":[],"kind":"article","role":"collectionGroup"},"doc://amplify-swift.Amplify/documentation/Amplify/AmplifyError":{"navigatorTitle":[{"kind":"identifier","text":"AmplifyError"}],"role":"symbol","title":"AmplifyError","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AmplifyError","abstract":[{"type":"text","text":"Amplifyâ€™s philosophy is to expose friendly error messages to the customer that assist with debugging."},{"type":"text","text":" "},{"type":"text","text":"Therefore, failable APIs are declared to return error results with Amplify errors, which require"},{"type":"text","text":" "},{"type":"text","text":"recovery suggestions and error messages."}],"kind":"symbol","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AmplifyError"}],"url":"\/documentation\/amplify\/amplifyerror"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/signedOut(_:_:_:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"signedOut"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier","text":"ErrorDescription"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","text":"RecoverySuggestion","preciseIdentifier":"s:7Amplify18RecoverySuggestiona"},{"kind":"text","text":", "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":"?)"}],"title":"AuthError.signedOut(_:_:_:)","kind":"symbol","type":"topic","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/signedOut(_:_:_:)","url":"\/documentation\/amplify\/autherror\/signedout(_:_:_:)","abstract":[{"text":"Caused when an operation needs the user to be in signedIn state","type":"text"}],"role":"symbol"},"doc://amplify-swift.Amplify/s28CustomDebugStringConvertibleP":{"title":"Swift.CustomDebugStringConvertible","type":"unresolvable","identifier":"doc:\/\/amplify-swift.Amplify\/s28CustomDebugStringConvertibleP"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/invalidState(_:_:_:)":{"fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"invalidState"},{"kind":"text","text":"("},{"text":"ErrorDescription","preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier"},{"kind":"text","text":", "},{"text":"RecoverySuggestion","preciseIdentifier":"s:7Amplify18RecoverySuggestiona","kind":"typeIdentifier"},{"kind":"text","text":", "},{"text":"Error","preciseIdentifier":"s:s5ErrorP","kind":"typeIdentifier"},{"text":"?)","kind":"text"}],"title":"AuthError.invalidState(_:_:_:)","kind":"symbol","type":"topic","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/invalidState(_:_:_:)","url":"\/documentation\/amplify\/autherror\/invalidstate(_:_:_:)","abstract":[{"text":"Caused when an operation is not valid with the current state of Auth category","type":"text"}],"role":"symbol"},"doc://amplify-swift.Amplify/documentation/Amplify":{"url":"\/documentation\/amplify","type":"topic","title":"Amplify","role":"collection","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify","kind":"symbol","abstract":[]},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/notAuthorized(_:_:_:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"notAuthorized"},{"kind":"text","text":"("},{"text":"ErrorDescription","preciseIdentifier":"s:7Amplify16ErrorDescriptiona","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:7Amplify18RecoverySuggestiona","text":"RecoverySuggestion"},{"kind":"text","text":", "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"text":"?)","kind":"text"}],"type":"topic","url":"\/documentation\/amplify\/autherror\/notauthorized(_:_:_:)","title":"AuthError.notAuthorized(_:_:_:)","kind":"symbol","identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/notAuthorized(_:_:_:)","role":"symbol","abstract":[{"text":"Caused when the current session is not authorized to perform an operation","type":"text"}]},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/AmplifyError-Implementations":{"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/AmplifyError-Implementations","abstract":[],"title":"AmplifyError Implementations","kind":"article","role":"collectionGroup","type":"topic","url":"\/documentation\/amplify\/autherror\/amplifyerror-implementations"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError":{"navigatorTitle":[{"kind":"identifier","text":"AuthError"}],"kind":"symbol","role":"symbol","url":"\/documentation\/amplify\/autherror","title":"AuthError","type":"topic","abstract":[{"type":"text","text":"Amplify error raised in the Auth category."}],"fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"text":"AuthError","kind":"identifier"}],"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError"},"doc://amplify-swift.Amplify/SQ":{"identifier":"doc:\/\/amplify-swift.Amplify\/SQ","type":"unresolvable","title":"Swift.Equatable"},"doc://amplify-swift.Amplify/s8CopyableP":{"title":"Swift.Copyable","type":"unresolvable","identifier":"doc:\/\/amplify-swift.Amplify\/s8CopyableP"},"doc://amplify-swift.Amplify/documentation/Amplify/AuthError/unknown(_:_:)":{"identifier":"doc:\/\/amplify-swift.Amplify\/documentation\/Amplify\/AuthError\/unknown(_:_:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"text":"unknown","kind":"identifier"},{"text":"(","kind":"text"},{"text":"ErrorDescription","kind":"typeIdentifier","preciseIdentifier":"s:7Amplify16ErrorDescriptiona"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP","text":"Error"},{"kind":"text","text":"?)"}],"title":"AuthError.unknown(_:_:)","type":"topic","url":"\/documentation\/amplify\/autherror\/unknown(_:_:)","kind":"symbol","abstract":[{"type":"text","text":"Caused by an unknown reason"}],"role":"symbol"}}}